src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5263,12 +5264,58 @@ private BoundExpression BindCollectionExpression(CollectionExpressionSyntax synt
 
             MessageID.IDS_FeatureCollectionExpressions.CheckFeatureAvailability(diagnostics, syntax, syntax.OpenBracketToken.GetLocation());
 
+            BoundUnconvertedWithElement? withElement = null;
+
             var builder = ArrayBuilder<BoundNode>.GetInstance(syntax.Elements.Count);
             foreach (var element in syntax.Elements)
             {
-                builder.Add(bindElement(element, diagnostics, this, nestingLevel));
+                if (element is WithElementSyntax withElementSyntax)
+                {
+                    // Report a withElement that is not first. Note: for the purposes of error recovery and diagnostics
+                    // we still bind the arguments in those later with elements.  However, we only validate those
+                    // arguments against the final arguments against the destination target type if the with element
+                    // was in the proper position.
+
+                    var analyzedArguments = AnalyzedArguments.GetInstance();
+
+                    // PROTOTYPE: Spec says we should only allow arglist if trivial.  Circle back on this and see if
+                    // this just falls out with the 'allowArgList: true' below.  If so, let LDM know it was easy and
+                    // allow it.  If it requires substantial work beyond this, disallow it for this feature.
+                    BindArgumentsAndNames(withElementSyntax.ArgumentList, diagnostics, analyzedArguments, allowArglist: true);
+
+                    var arguments = analyzedArguments.Arguments;
+                    for (int i = 0; i < arguments.Count; i++)
+                    {
+                        var arg = arguments[i];
+                        if (arg.Type is { TypeKind: TypeKind.Dynamic })
+                        {
+                            diagnostics.Add(ErrorCode.ERR_CollectionArgumentsDynamicBinding, arg.Syntax);
+                            arguments[i] = new BoundBadExpression(arg.Syntax, LookupResultKind.Empty, symbols: [], childBoundNodes: [arg], type: Compilation.GetSpecialType(SpecialType.System_Object));
+                        }
+                    }
+
+                    if (withElementSyntax != syntax.Elements.First())
+                    {
+                        diagnostics.Add(ErrorCode.ERR_CollectionArgumentsMustBeFirst, withElementSyntax.WithKeyword);
```

Comment:
It looks like we're dropping bound nodes

Reply:
is that a problem?  i genuinely don't know?  I could have an InvalidWithElements array on the collection node.  What is your preference on this sort of thing?

Reply:
i followed the approach taken with a bad `new(...)` call now.

Reply:
Yes, there are multiple problems if bound nodes are dropped:
- the semantic model will lack context for that syntax (scenario: QuickInfo on the argument of a `with` in incorrect position). This will also be a problem in IOperation tree
- downstream analysis passes such as definite assignment or nullability analysis will lack context on things we definitely understood/bound

Reply:
Ok.  Have added those to the tree.